src/Service/CaptureService.php

```diff
@@ -134,46 +137,32 @@ public function doOpenInvoiceCapture(string $orderNumber, $captureAmount, Contex
                 }
 
                 $lineItems = $order->getLineItems();
-                $lineItemsArray = $this->getLineItemsArray($lineItems, $order->getCurrency()->getIsoCode());
-
-                $additionalData = array_merge($lineItemsArray, [
-                    'openinvoicedata.shippingCompany' => $delivery->getShippingMethod()->getName(),
-                    'openinvoicedata.trackingNumber' => $delivery->getTrackingCodes(),
-                ]);
+                $lineItemsObjectArray = $this->getLineItemsObjectArray($lineItems);
 
                 $request = $this->buildCaptureRequest(
-                    $customFields[PaymentResponseHandler::ORIGINAL_PSP_REFERENCE],
                     $captureAmount,
                     $currencyIso,
                     $order->getSalesChannelId(),
-                    $additionalData
-                );
-
-                $this->clientService->logRequest(
-                    $request,
-                    Client::API_PAYMENT_VERSION,
-                    '/pal/servlet/Payment/{version}/capture',
-                    $order->getSalesChannelId()
+                    $lineItemsObjectArray
                 );
 
-                $response = $this->sendCaptureRequest($client, $request);
-
-                $this->clientService->logResponse(
-                    $response,
-                    $order->getSalesChannelId()
+                $response = $this->sendCaptureRequest(
+                    $client,
+                    $customFields[PaymentResponseHandler::ORIGINAL_PSP_REFERENCE],
+                    $request
                 );
 
-                if ('[capture-received]' === $response['response']) {
+                if ($response->getStatus() == "received") {
                     $this->saveCaptureRequest(
                         $orderTransaction,
-                        $response['pspReference'],
+                        $response->getPspReference(),
                         PaymentCaptureEntity::SOURCE_SHOPWARE,
                         PaymentCaptureEntity::STATUS_PENDING_WEBHOOK,
                         intval($captureAmount),
                         $context
                     );
                 }
-                $results[] = $response;
+                $results[] = $response->__toString();
```

Comment:
Response array should be assigned to `results`, however `__toString()` returns a json encoded value.